lightkurve/seismology/core.py

```diff
@@ -543,6 +543,12 @@ def estimate_numax(self, method="acf2d", **kwargs):
         method = validate_method(method, supported_methods=["acf2d"])
         if method == "acf2d":
             from .numax_estimators import estimate_numax_acf2d
+
+            fs = np.mean(np.diff(self.periodogram.frequency))
+            if not np.isclose(np.median(np.diff(self.periodogram.frequency.value)), fs.value):
```

Comment:
I'm confused by this, what is the logic behind comparing
`np.mean(np.diff(self.periodogram.frequency)).value`
against
`np.median(np.diff(self.periodogram.frequency.value))` ?

Reply:
If the frequencies are uniformly spaced, the first differences of the periodogram frequency should all be (near) identical. In this scenario, the mean and the median of the first differences should also be near identical. If the frequencies are not evenly spaced, the mean and the median of the differences will begin to diverge.

I'm not sure if this is the most robust way of checking this. Basically it just needs a simple operation that checks whether there is uniformity of values in an array--- you might have a more elegant solution!

Reply:
Ahh, I got distracted by the repeated call to `np.diff` and the different location of `.value`, which made me think something more mysterious was happening.  If you want to use this solution, it would be best to document and simplify the code as follows:
```python
# Detect whether the frequency grid is evenly-spaced
# by comparing the median and mean of the first differences
freqdiff = np.diff(self.periodogram.frequency.value)
if not np.isclose(np.median(freqdiff), np.mean(freqdiff)):
```
*However* I think you can further simplify this as follows:
```python
# Detect whether the frequency grid is evenly-spaced
# by verifying that the differences are all equal
freqdiff = np.diff(self.periodogram.frequency.value)
if not np.allclose(freqdiff[0], freqdiff):
```

Does that make sense?

To quickly run the test locally after making this change, you can type `pytest lightkurve/seismology/tests/test_butler.py --remote-data`.

Reply:
I knew you'd have a more elegant solution! I've checked and tested this works!